File: backend/internal/database/models/slide.model.go
package models

import "github.com/gianghp123/Vidmerce/backend/internal/core/enums"

type SlideEntity struct {
	ImageRole enums.ImageRole `dynamodbav:"imageRole"`
	// Overlay text on image
	OverlayText *string `dynamodbav:"overlayText,omitempty"`
	// S3 key for image
	S3Key string `dynamodbav:"s3Key"`
	// SlideEntity number
	SlideNumber int `dynamodbav:"slideNumber"`
}

File: frontend/src/lib/models/slide.model.ts
import type { ImageRole } from "../enums/image-role.enum";

export interface ISlide {
  imageRole: ImageRole;
  /**
   * Overlay text on image
   */
  overlayText?: string;
  /**
   * S3 key for image
   */
  s3Key: string;
  /**
   * ISlide number
   */
  slideNumber: number;
}
